orm/src/test/java/org/hibernate/search/test/util/ResourceCleanupFunctions.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.test.util;
+
+import java.util.function.Consumer;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.EntityTransaction;
+
+import org.hibernate.search.jpa.FullTextEntityManager;
+import org.hibernate.search.jpa.Search;
+
+/**
+ * Some helpers to allow using lamda blocks to ensure that
+ * resources which get opened are also being closed correctly.
+ *
+ * @author Sanne Grinovero (C) 2016 Red Hat Inc.
+ */
+public final class ResourceCleanupFunctions {
+
+	private ResourceCleanupFunctions() {
+		//Utility class: not to be constructed.
+	}
+
+	/**
+	 * Given an EntityManagerFactory, we'll open an EntityManager, allow some
+	 * code block to use it as a FullTextEntityManager, and finally close
+	 * the EntityManager.
+	 * @param emf The EntityManagerFactory which we'll use to create a new EntityManager
+	 * @param consumer A block of code which uses the FullTextEntityManager
+	 */
+	public static void withinEntityManager(EntityManagerFactory emf, Consumer<FullTextEntityManager> consumer) {
+		EntityManager entityManager = emf.createEntityManager();
+		try {
+			FullTextEntityManager fem = Search.getFullTextEntityManager( entityManager );
+			consumer.accept( fem );
+		}
+		finally {
+			entityManager.close();
+		}
```

Comment:
I think we should handle exceptions with more caution. Instead of `finally`, this would be:

```
catch (Throwable t1) {
	try {
		entityManager.close();
	}
	catch(Throwable t2) {
		someLogger.warn("Error while automatically closing an entity manager", e); // Maybe we could remove that
		t1.addSuppressed(t2);
	}
	throw t1;
}
entityManager.close(); // Let any exception when closing propagate, since we didn't catch any exception before
```